### DISCUSSION TITLE
Use .gitignore for large files?

### DISCUSSION THREAD

**Discussion by zlkvet** [community/community#32263] [New to GitHub] [answered]
I am working with large .tif files (rasters) that need to be in the R project folder for the code to work, but they seem to prevent the repo from being able to push because of the file size (some over 2gb). These large files do not really need to be saved to the repo because they are freely available to download if ever lost, but I need to keep them in the folder. I am looking for a work around so that I can keep all the files I want in the local folder associated with my repo, but have some be overlooked when adding/committing/pushing so that they don't slow down the process as much. 
I have tried adding /.tif to my gitignore file, but that doesn't seem to work. Any help appreciated.

**Comment by averagebit** [comment, marked as answer]
Hi @zlkvet 

What you're trying to achieve is certainly possible, however, I wouldn't be able to tell you how to do it precisely without knowing your current folder/file structure and names. Instead you can take a look at [this](https://www.atlassian.com/git/tutorials/saving-changes/gitignore) and/or [this](https://git-scm.com/docs/gitignore) guide about how the gitignore file works.

As a short example if the file name is `foo.tif` and the folder name the file is in is called `bar` you can put `bar/foo.tif` in your gitignore file and that should ignore the file but not the folder. In case you wanted to ignore all the files ending in `.tif` in the `bar` folder you could add `bar/*.tif` in your gitignore. However, in case you already have the file committed to the repository you might need to delete it from the git history before the gitignore changes take effect on future commits.

**Reply by zlkvet** [reply]
Thanks! In the end, I did just delete the repo and local repo to start from scratch, making sure to update my .gitignore before the first commit. I used *.tif and *.shp, plus a few other file types added so that no matter where they are in the repo, they are ignored. It appears to be working for now! :)

**Comment by airtower-luna** [comment]
`.gitignore` doesn't apply to files that are already committed, you need to remove the files from your repository before it'll take effect. The tricky thing is that committing a remove (you can do that with [`git rm --cached`](https://git-scm.com/docs/git-rm#Documentation/git-rm.txt---cached) without deleting the actual files) isn't going to be enough: The files are already part of your repository history, so they'll still get pushed, and GitHub will reject the push.

You have two options:

* [Remove the files from history](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github#removing-files-from-a-repositorys-history), too.
* If you don't mind permanently losing :warning: the existing Git history, you can delete and recreate the *local* repository (the `.git` directory), and never commit the large files in the first place.

**Comment by EmptyDeck** [comment]
Try this

https://github.com/EmptyDeck/Large-File-Ignorer-for-Github
